poc_mobile_verifier/src/heartbeats.rs

```diff
@@ -154,12 +154,16 @@ struct HeartbeatSaveResult {
     inserted: bool,
 }
 
+#[derive(thiserror::Error, Debug)]
+#[error(transparent)]
+pub struct SaveHeartbeatError(#[from] sqlx::Error);
+
 impl Heartbeat {
     pub async fn validate_heartbeats<'a>(
         heartbeats: impl Stream<Item = CellHeartbeat> + 'a,
         epoch: &'a Range<DateTime<Utc>>,
-    ) -> Result<impl Stream<Item = Self> + 'a> {
-        Ok(heartbeats.map(move |heartbeat_report| {
+    ) -> impl Stream<Item = Self> + 'a {
```

Comment:
By removing this result wrapper are we saying that, while processing a stream’s elements may fail in some fashion, actually constructing the stream is a pure functional operation that can’t fail? If so that would make sense to me.

Reply:
yup!